nipoppy/workflows/bids_conversion.py

```diff
@@ -50,12 +51,17 @@ def dpaths_to_check(self) -> list[Path]:
 
     @cached_property
     def pipeline_config(self) -> BidsPipelineConfig:
-        """Get the user config for the BIDS conversion software."""
+        """Get the user config for the BIDS conversion pipeline."""
         return self.config.get_pipeline_config(
             self.pipeline_name,
             self.pipeline_version,
         )
 
+    @cached_property
+    def pipeline_step_config(self) -> BidsPipelineStepConfig:
+        """Get the config for the relevant step of the BIDS conversion pipeline."""
+        return super().pipeline_step_config
+
```

Comment:
It was to modify the type annotation for the return value. There might be a better way to do this, e.g. [with generic types](https://stackoverflow.com/a/61239475)

Reply:
> It was to modify the type annotation for the return value. There might be a better way to do this, e.g. [with generic types](https://stackoverflow.com/a/61239475)

Looked into this a bit. Not sure it would work very well because of the class hierarchy:
```
BasePipelineWorkflow
├── PipelineRunner
│   └── BidsConversionRunner
└── PipelineTracker
```

Where `PipelineRunner` and `PipelineTracker` use `ProcPipelineConfig`/`ProcPipelineStepConfig` while `BidsConversionRunner` uses `BidsPipelineConfig`/`BidsPipelineStepConfig`.

If we make `BasePipelineWorkflow` a generic class, then set the types in `PipelineRunner`, then there is no way to override these types in `BidsConversionRunner`...